### PR TITLE
ipm: new package

### PR DESCRIPTION
**Pull request by Christoph-TU**
Depends on #39881 

This is the package for the MPI profiler IPM. 

IPM comes with a Perl script which parses the generated XML files during profiling. For this script, I added an optional runtime dependency on Ploticus, for which I already wrote a package here #39881. Since Ploticus is not widely used anymore, there might be another way to include this dependency without creating its own package. However, this dependency is not crucial on its own, and it might also be an option to leave the installation of Ploticus to the end user.

### CODE REVIEW

File: var/spack/repos/builtin/packages/ipm/package.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+from spack.util.executable import Executable
+
+
+class Ipm(AutotoolsPackage):
+    """IPM is a portable profiling infrastructure for parallel codes.
+    It provides a low-overhead profile of application performance
+    and resource utilization in a parallel program. Communication,
+    computation, and IO are the primary focus."""
+
+    homepage = "https://github.com/nerscadmin/IPM"
```
Comment (Christoph-TU) [inline var/spack/repos/builtin/packages/ipm/package.py:16]: The GitHub page of IPM links to http://ipm-hpc.org/, but this page currently can't be reached. However, there seems to be an old version of the website available at https://ipm-hpc.sourceforge.net/ which provides some information about IPM.

File: var/spack/repos/builtin/packages/ipm/package.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    depends_on("libunwind", when="+libunwind")
+
+    # These are required when running the perl script ipm_parse,
+    # which is used to create reports from the generated xml file
+    depends_on("perl", type="run", when="+parser")
+    depends_on("ploticus", type="run", when="+parser")
+
+    # 2COMPLEX and 2DOUBLE_COMPLEX are non-standard types and lead
+    # to compile errors when building with coll_details
+    patch("remove_MPI_2COMPLEX_and_MPI_2DOUBLE_COMPLEX.patch", when="+coll_details")
```
Comment (Christoph-TU) [inline var/spack/repos/builtin/packages/ipm/package.py:57]: I couldn't determine exactly when MPI_2COMPLEX and MPI_2DOUBLE_COMPLEX were removed from the standard, so I removed them regardless of the MPI version. On another note, Open MPI still supports these two datatypes, but I could not find a way to include this condition in the 'when' clause.

**Review by tldahlgren — APPROVED** [review]
Confirmed the homepage, branch, and tag exist.

File: var/spack/repos/builtin/packages/ipm/package.py (hunk 2 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/ipm/package.py:57]: Unfortunately I don't think we have a way to indicate the equivalent of "when not `^openmpi`.

**Comment by tldahlgren** [comment]
Audit checks indicate you need to add another package:

```
1m 51s
Run . share/spack/setup-env.sh
PKG-DIRECTIVES: 2 issues found
1. ipm: unknown package 'ploticus' in 'depends_on' directive
     in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/ipm/package.py
2. ipm: dependency on ploticus cannot be satisfied by known versions of ploticus
    happening in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/ipm/package.py
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Package requires the addition of another package for `ploticus`.

**Comment by Christoph-TU** [comment]
@spackbot re-run pipeline

**Comment by Christoph-TU** [comment]
@spackbot commands

**Comment by Christoph-TU** [comment]
@spackbot rebuild everything

**Comment by Christoph-TU** [comment]
@tldahlgren Ploticus is now present in the development branch.

**Review by tldahlgren — APPROVED** [review]
I'm able to successfully build the package.

**Pull request merged**